src/NuGet.Core/NuGet.Common/MsBuildStringUtility.cs

```diff
@@ -112,5 +112,36 @@ public static string Convert(string value)
 
             return value.Replace(',', ';');
         }
+
+        /// <summary>
+        /// Return empty list of NuGetLogCode if all lists of NuGetLogCode are not the same. 
+        /// </summary>
+        public static IEnumerable<NuGetLogCode> GetSingleOrDefaultDistinctNuGetLogCodes(IEnumerable<IEnumerable<NuGetLogCode>> nugetLogCodes)
+        {
+            if (nugetLogCodes.Count() > 0)
+            {
+                var result = Enumerable.Empty<NuGetLogCode>();
+                var first = true;
+
+                foreach (var logCode in nugetLogCodes)
+                {
+                    // If this is first item, assign it to result
+                    if (first)
+                    {
+                        result = logCode;
+                        first = false;
+                    }
+                    // Compare the rest items to the first one.
+                    else if (result == null || result.Count() != logCode.Count() || !result.All(logCode.Contains))
```

Comment:
1. When will `result` be null?
2. Use `EqualityUtility.SequenceEqualWithNullCheck` or `EqualityUtility.SetEqualsWithNullCheck` for comparison